cocotb/result.py

```diff
@@ -86,7 +86,12 @@ def create_error(obj, msg):
 
 class ReturnValue(Exception):
     """Helper exception needed for Python versions prior to 3.3."""
+
     def __init__(self, retval):
+        warnings.warn(
+            "``ReturnValue`` is deprecated, use a normal return statement instead.",
+            DeprecationWarning,
+            stacklevel=2)
```

Comment:
Can you add a test somewhere of this warning?

Reply:
Something like:
```python
@cocotb.test()
async def test_returnvalue_deprecated(dut):
	@cocotb.coroutine
	def get_value():
	    yield Timer(1, units='ns')
	    raise ReturnValue(42)

	with warnings.catch_warnings(record=True) as w:
	    # Cause all warnings to always be triggered.
	    warnings.simplefilter("always")
	    val = await get_value()
	    assert val == 42
	    assert len(w) == 1
	    assert issubclass(w[-1].category, DeprecationWarning)
	    assert "deprecated" in str(w[-1].message)
```

Reply:
Added this to a new file specifically for testing deprecated features.